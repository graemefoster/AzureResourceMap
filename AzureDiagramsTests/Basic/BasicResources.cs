using Shouldly;

namespace AzureDiagramsTests.Basic;

public class BasicResources
{
    [Fact]
    public async Task SingleStorageAccount()
    {
        var resources = TestResourcesObjectMother
            .WithPublicAccessibleStorageAccount();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources.ToArray(),
            false, 
            false, 
            false, 
            false,
            false);
        
        diagram.ShouldMatchApproved();
    }
}