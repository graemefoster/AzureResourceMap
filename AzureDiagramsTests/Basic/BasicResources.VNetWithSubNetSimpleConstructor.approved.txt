<mxGraphModel>
	<root>
		<mxCell id="0" />
		<mxCell id="1" parent="0" />
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba.subnet1.empty" value="Empty&lt;br/&gt;(Subnet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#ffffff;verticalAlign=middle" vertex="1" parent="da6247fd-0679-ddbd-3294-b327594e85ba.subnet1">
    <mxGeometry x="50" y="50" width="150" height="75" 
    as="geometry" />
</mxCell>
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba.subnet2.empty" value="Empty&lt;br/&gt;(Subnet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#ffffff;verticalAlign=middle" vertex="1" parent="da6247fd-0679-ddbd-3294-b327594e85ba.subnet2">
    <mxGeometry x="50" y="50" width="150" height="75" 
    as="geometry" />
</mxCell>
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba.subnet1" value="subnet1&lt;br/&gt;(10.1.0.0/26)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="da6247fd-0679-ddbd-3294-b327594e85ba">
    <mxGeometry x="50" y="50" width="250" height="185" 
    as="geometry" />
</mxCell>
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba.subnet2" value="subnet2&lt;br/&gt;(10.1.1.0/26)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="da6247fd-0679-ddbd-3294-b327594e85ba">
    <mxGeometry x="400" y="50" width="250" height="185" 
    as="geometry" />
</mxCell>
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba" value="vnet&lt;br/&gt;(VNet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F8CECC;verticalAlign=top" vertex="1" parent="1">
    <mxGeometry x="0" y="0" width="700" height="295" 
    as="geometry" />
</mxCell>
<mxCell id="da6247fd-0679-ddbd-3294-b327594e85ba.image.0" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="da6247fd-0679-ddbd-3294-b327594e85ba">
    <mxGeometry x="660" y="255" width="30" height="30" 
    as="geometry" />
</mxCell>


	</root>
</mxGraphModel>